llm/codellama/README.md

```diff
@@ -0,0 +1,185 @@
+ GCP     a2-highgpu-8g               96      680       A100:8         us-central1-a   29.39               
+ AWS     p4d.24xlarge                96      1152      A100:8         us-east-1       32.77               
+ Azure   Standard_ND96amsr_A100_v4   96      1924      A100-80GB:8    eastus          32.77               
+ GCP     a2-ultragpu-8g              96      1360      A100-80GB:8    us-central1-a   40.22               
+ AWS     p4de.24xlarge               96      1152      A100-80GB:8    us-east-1       40.97               
+----------------------------------------------------------------------------------------------------------
+
+Launching a cluster 'code-llama'. Proceed? [Y/n]: 
+```
+2. Send a request to the endpoint for code completion:
+```bash
+IP=$(sky status --ip code-llama)
+
+curl -L http://$IP:8000/v1/completions \
+    -H "Content-Type: application/json" \
+    -d '{
+      "model": "codellama/CodeLlama-70b-Python-hf",
+      "prompt": "def quick_sort(a: List[int]):",
+      "max_tokens": 256
+  }' | jq -r '.choices[0].text'
+```
+
+This returns the following completion:
+```python
+    if len(a) <= 1:
+        return a
+    pivot = a.pop(len(a)//2)
+    b = []
+    c = []
+    for i in a:
+        if i > pivot:
+            b.append(i)
+        else:
+            c.append(i)
+    b = quick_sort(b)
+    c = quick_sort(c)
+    res = []
+    res.extend(c)
+    res.append(pivot)
+    res.extend(b)
+    return res
+```
+
+## Scale up the service with SkyServe
+
+1. With [SkyServe](https://skypilot.readthedocs.io/en/latest/serving/sky-serve.html), scaling up the Code Llama service is as simple as running:
+```bash
+sky serve up -n code-llama ./endpoint.yaml
+```
+This will start the service with multiple replicas on the cheapest available locations and accelerators. SkyServe will automatically manage the replicas, monitor their health, and restart them when needed.
+
+A single endpoint will be returned and any request sent to the endpoint will be routed to the healthy replicas.
+
+2. To check the status of the service, run:
+```bash
+sky serve status code-llama
+```
+After a while, you will see the following output:
+```console
+Services
+NAME        VERSION  UPTIME  STATUS        REPLICAS  ENDPOINT            
+code-llama  1        -       READY         2/2       3.85.107.228:30002  
+
+Service Replicas
+SERVICE_NAME  ID  VERSION  IP  LAUNCHED    RESOURCES                   STATUS REGION  
+code-llama    1   1        -   2 mins ago  1x Azure({'A100-80GB': 2})) READY  eastus  
+code-llama    2   1        -   2 mins ago  1x GCP({'L4': 8}))          READY  us-east4-a 
+```
+As shown, the service is now backed by 2 replicas, one on Azure and one on GCP, and the accelerator
+type is chosen to be **the cheapest and available one** on the clouds. That said, it maximizes the
+availability of the service while minimizing the cost.
```

Comment:
What does it mean by 'maximize availability' here

Reply:
More like it increases the chance of getting the resources with all the candidate resources in multiple resource pools.